src/System.Net.WebSockets.Client/tests/ClientWebSocketUnitTest.cs

```diff
@@ -1,39 +1,45 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Net.WebSockets;
 using System.Threading;
 using System.Threading.Tasks;
-using System.Text;
 
 using Xunit;
+using Xunit.Abstractions;
 
-namespace System.Net.WebSockets.Client.Test
+namespace System.Net.WebSockets.Client.Tests
 {
     /// <summary>
     /// ClientWebSocket unit tests that do not require a remote server.
     /// </summary>
-    public class ClientWebSocketTest
+    public class ClientWebSocketUnitTest
     {
-        internal static class AssertExtensions
+        private readonly ITestOutputHelper _output;
+
+        public ClientWebSocketUnitTest(ITestOutputHelper output)
         {
-            public static void Throws<T>(Action action, string message)
-                where T : Exception
-            {
-                Assert.Equal(Assert.Throws<T>(action).Message, message);
-            }
+            _output = output;
         }
 
         [Fact]
-        public void ClientWebSocket_Ctor()
+        public void Ctor_Success()
         {
+            if (WebSocketHelper.ShouldSkipTestOSNotSupported(_output))
+            {
+                return;
+            }
+
```

Comment:
I've never actually looked into this until now, but seeing each of these tests begin with this boilerplate made me want to see if xunit extensibility could be employed here.  Instead of doing this, I just tried adding this code to the project:
```C#
[XunitTestCaseDiscoverer("System.Net.WebSockets.Client.Tests.RequiresWebSocketsFactDiscoverer", "System.Net.WebSockets.Client.Tests")]
[AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
public class RequiresWebSocketsFact : FactAttribute { }

public class RequiresWebSocketsFactDiscoverer : IXunitTestCaseDiscoverer
{
    private readonly IMessageSink _diagnosticMessageSink;

    public RequiresWebSocketsFactDiscoverer(IMessageSink diagnosticMessageSink)
    {
        _diagnosticMessageSink = diagnosticMessageSink;
    }

    public IEnumerable<IXunitTestCase> Discover(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
    {
        return WebSocketsHelper.WebSocketsIsSupported ?
            new IXunitTestCase[] { new XunitTestCase(_diagnosticMessageSink, discoveryOptions.MethodDisplayOrDefault(), testMethod, null) } :
            Array.Empty<IXunitTestCase>();
    }
}
```
and then instead of this boilerplate at the beginning of each method, I was able to attribute each test with:
```C#
[RequiresWebSocketsFact]
```
instead of:
```C#
[Fact]
```
Something to consider.